File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.cc
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>

#define MAXN 1000000
char prime[MAXN + 1];

void sieve(void) {
    for (int i = 2; i <= MAXN; ++i) prime[i] = 1;
    for (int i = 2; i * i <= MAXN; ++i) {
        if (prime[i]) {
            for (int j = i * i; j <= MAXN; j += i)
                prime[j] = 0;
        }
    }
}

int main(void) {
    sieve();

    int N;
    while (scanf("%d", &N) == 1 && N != 0) {
        int found = 0;
        for (int a = 3; a <= N - a; a += 2) {
            if (prime[a] && prime[N - a]) {
                printf("%d = %d + %d\n", N, a, N - a);
                found = 1;
                break;
            }
        }
        if (!found) {
            printf("Goldbach's conjecture is wrong.\n");
        }
    }
    return 0;
}

File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.py
import sys
input = sys.stdin.readline

LIMIT = 1000000

is_prime = [True] * (LIMIT + 1)
is_prime[0] = is_prime[1] = False
for p in range(2, int(LIMIT ** 0.5) + 1):
    if is_prime[p]:
        for j in range(p * p, LIMIT + 1, p):
            is_prime[j] = False

while True:
    N = int(input())
    if N == 0:
        break

    found = False
    for a in range(3, N // 2 + 1, 2):
        if is_prime[a] and is_prime[N - a]:
            print("%d = %d + %d" % (N, a, N - a))
            found = True
            break

    if not found:
        print("Goldbach's conjecture is wrong.")
